src/mpdecode_core.c

```diff
@@ -79,7 +79,7 @@ void encode(struct LDPC *ldpc, unsigned char ibits[], unsigned char pbits[]) {
 
         for (i=0; i<ldpc->max_row_weight; i++) {
             ind = H_rows[p + i*ldpc->NumberParityBits];
-            par = par + ibits[ind-1];
+            if (ind) par = par + ibits[ind-1];
```

Comment:
@billcowley :point_up:

Reply:
Good show David!  Don't know why this did not cause a problem before since many of the codes use irregular weights. Having seen the H_rows and H_cols structures I'll try to look in to this area in a few days just to understand why it's working - from your comments yesterday, maybe two effects are cancelling each other out(?)   Does the C code now give the same performance as the octave?  ie steep waterfall up to 2dB then error floor slightly above 1e-6?

Reply:
Thanks @billcowley - I looked at a few other codes and they seem to use all entries on each row.  Yes the (8192,4096) code is performing as expected in the unit tests - 1.5 to 2 dB better than the shorter rate 0.5 codes :smile: 

```
$ ./src/ldpc_enc /dev/zero - --code H_4096_8192_3d --testframes 100 --sd | ./src/ldpc_noise - - 1.5 | ./src/ldpc_dec - /dev/null --code H_4096_8192_3d --testframes --sd
Uncoded PSK Eb/No simulation:
No    =  1.50 dB (1.41 linear)
Eb    =  0.00 dB (1.00 linear)
Eb/No = -1.50 dB (0.71 linear)
Using: H_4096_8192_3d
Using: H_4096_8192_3d
Nframes: 100
CodeLength: 8192 offset: 0
.........x.x............................x......................................................written: 819200
..measured double sided (real) noise power: 0.705781
x..total iters 9102
Raw   Tbits: 819200 Terr:  95516 BER: 0.117
Coded Tbits: 409600 Terr:     15 BER: 0.000
      Tpkts:    100 Tper:      4 PER: 0.040
```

This is the form we use for the unit testing, simple BPSK simulation - I think Eb/No is raw/uncoded.

At 2dB coded Eb/No it is getting 1% PER, not sure about the coded BER, not enough digits to see 1E-6!  Will fix that soon.

Currently working on integration with the FreeDV API data modes.